File: docs/examples/python/remove_whitelist.py
"""Delete an IP address from the whitelist."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"


def remove_whitelist() -> None:
    response = requests.delete(
        f"{BASE_URL}/settings/whitelist/192.168.0.10",
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    print("Removed:", response.json())


if __name__ == "__main__":
    remove_whitelist()

File: docs/examples/go/remove_whitelist.go
// Command remove_whitelist deletes an IP from the whitelist.
package main

import (
    "encoding/json"
    "fmt"
    "net/http"
)

const whitelistRemovalToken = "<token>"

func main() {
    req, _ := http.NewRequest(http.MethodDelete, baseURL+"/settings/whitelist/192.168.0.10", nil)
    req.Header.Set("Authorization", "Bearer "+whitelistRemovalToken)

    resp, err := http.DefaultClient.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        panic(fmt.Sprintf("unexpected status: %d", resp.StatusCode))
    }

    var result struct {
        Success bool `json:"success"`
    }
    if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
        panic(err)
    }

    fmt.Println("Removal status:", result.Success)
}
